src/dndbeyond/base/utils.js

```diff
@@ -223,6 +223,19 @@ async function buildAttackRoll(character, attack_source, name, description, prop
                 crit_damages[0] = damagesToCrits(character, ["8d12"])[0];
             if (roll_properties.name === "Jim’s Magic Missile")
                 crit_damages[0] = damagesToCrits(character, ["3d4"])[0];
+            const matchViciousWeapon = description.match(/(When you roll a 20 on your attack roll with this magic weapon, the target takes an extra )(\d+)( damage of the weapon’s type.)/);
+            if (matchViciousWeapon) {
+                let vicious_damage_present = false;
+                damages.forEach(function (damage, index) {
+                    if (damage == matchViciousWeapon[2]) {
+                        vicious_damage_present = true;
+                    }
+                });
+                if (!vicious_damage_present) {
```

Comment:
Here's a better way to do the check... instead of a temp variable and using `forEach`, you can do : 
```js
if (damages.some(damage => damage === matchViciousWeapon[2])) {
}
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some